runtime/parachains/src/dmp.rs

```diff
@@ -386,21 +386,7 @@ impl<T: Config> Pallet<T> {
 		let mut total_weight = T::DbWeight::get().reads_writes(1, 0);
 
 		// Bail out early if the queue is empty.
-		if queue_length == 0 {
-			return total_weight
-		}
-
-		// A call to [`check_processed_downward_messages`] will check if `processed_downward_messages`
-		// is greater than total messages in queue. so we don't need to check that again here.
-
-		if processed_downward_messages == 0 {
-			// This should never happen in practice, because of the advancement rule - parachains must
-			// process one message at least per para block.
```

Comment:
We no longer enforce this?

Reply:
We still enforce this, but this is not the place. The warn and debug assert dont make sense as this can be called when there are no messages to process and the log is spammed.

Reply:
[the check is here ](https://github.com/paritytech/polkadot/blob/sandreim/dmp_partial_processing_3/runtime/parachains/src/dmp.rs#L342)

Reply:
[called from here](https://github.com/paritytech/polkadot/blob/53253de77f1649ecbb650df2ec3e29d0d1110a2d/runtime/parachains/src/inclusion/mod.rs#L1065)